include/sqlpp11/insert_value_list.h

```diff
@@ -142,27 +148,44 @@ namespace sqlpp
   SQLPP_PORTABLE_STATIC_ASSERT(assert_insert_dynamic_set_statement_dynamic_t,
                                "dynamic_set must not be called in a static statement");
 
+  template <typename Expr>
+  struct must_not_insert
+  {
+	  static const bool value = must_not_insert_t<lhs_t<Expr>>::value;
+  };
+
   template <typename... Assignments>
   using check_insert_set_t = static_combined_check_t<
-      static_check_t<logic::all_t<is_assignment_t<Assignments>::value...>::value, assert_insert_set_assignments_t>,
-      static_check_t<not detail::has_duplicates<lhs_t<Assignments>...>::value, assert_insert_set_no_duplicates_t>,
-      static_check_t<logic::none_t<must_not_insert_t<lhs_t<Assignments>>::value...>::value,
+      static_check_t<logic::all_t<detail::is_assignment_impl<Assignments>::type::value...>::value, assert_insert_set_assignments_t>,
+      static_check_t<not detail::has_duplicates<typename lhs<Assignments>::type...>::value, assert_insert_set_no_duplicates_t>,
```

Comment:
Ugh. MSVC really struggles with this one? Again, a reference to the corresponding issue at MS would be really helpful, because I would be tempted to change that back every single time I saw it.